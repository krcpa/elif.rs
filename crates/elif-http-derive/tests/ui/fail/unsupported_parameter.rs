//! Test that handlers with unsupported parameters produce clear compile-time errors

use elif_http_derive::{get, controller, param};

// Mock the required types
pub struct ElifRequest;
pub struct ElifResponse;
pub type HttpResult<T> = Result<T, Box<dyn std::error::Error>>;

#[controller("/api/users")]
pub struct UserController;

impl UserController {
    // This should fail because `extra_param` is neither a path parameter nor ElifRequest
    #[get("/{id}")]
    #[param(id: int)]
    pub async fn show(&self, id: i32, extra_param: String, req: ElifRequest) -> String {
        format!("User ID: {}, Extra: {}", id, extra_param)
    }
}

fn main() {}