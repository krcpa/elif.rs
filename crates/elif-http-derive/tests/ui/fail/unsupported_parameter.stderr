error: Unsupported parameter 'extra_param' of type 'String'. Only path parameters (specified in route), body parameters (annotated with #[body]), and ElifRequest are supported. Hint: Remove this parameter, add it to the route path like '/users/{extra_param}' and annotate with #[param(extra_param: type)], annotate with #[body(extra_param: Type)], or use #[request] to enable automatic request injection.
  --> tests/ui/fail/unsupported_parameter.rs:17:39
   |
17 |     pub async fn show(&self, id: i32, extra_param: String, req: ElifRequest) -> String {
   |                                       ^^^^^^^^^^^^^^^^^^^

warning: unused import: `param`
 --> tests/ui/fail/unsupported_parameter.rs:3:41
  |
3 | use elif_http_derive::{get, controller, param};
  |                                         ^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
